pinot-query-planner/src/main/java/org/apache/pinot/query/routing/StageMetadata.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.routing;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.pinot.core.routing.TimeBoundaryInfo;
+
+
+/**
+ * {@code StageMetadata} is used to send stage-level info about the execution.
+ */
+public class StageMetadata {
+  private final List<WorkerMetadata> _workerMetadataList;
+  private final Map<String, String> _customProperties;
+
+  public StageMetadata(List<WorkerMetadata> workerMetadataList, Map<String, String> customProperties) {
+    _workerMetadataList = workerMetadataList;
+    _customProperties = customProperties;
+  }
+
+  public List<WorkerMetadata> getWorkerMetadataList() {
+    return _workerMetadataList;
+  }
+
+  public Map<String, String> getCustomProperties() {
+    return _customProperties;
+  }
+
+  public static class Builder {
```

Comment:
i am inclinde to use Builder/Extractor mechanism b/c we can always extend & modify the Builder/Extractor easily without worry of changing the underlying StageMetadata format going forward.